src/sql/workbench/services/notebook/common/sqlSessionManager.ts

```diff
@@ -270,9 +274,17 @@ export class SQLFuture extends Disposable implements FutureInternal {
 	private ioHandler: nb.MessageHandler<nb.IIOPubMessage>;
 	private doneHandler: nb.MessageHandler<nb.IShellMessage>;
 	private doneDeferred = new Deferred<nb.IShellMessage>();
+	private configuredMaxRows: number = MAX_ROWS;
 
-	constructor(private _queryRunner: QueryRunner, private _executionCount: number | undefined) {
+	constructor(private _queryRunner: QueryRunner, private _executionCount: number | undefined, private configurationService: IConfigurationService) {
 		super();
+		let config = configurationService.getValue(NotebookConfigSectionName);
```

Comment:
Minor: might want to update this value before running a query. Users expect changed to happen immediately if possible and not require reopening an editor.

Reply:
Wait, doesn't this happen already? I had a notebook open, changed/saved the setting, went back to the notebook, and the notebook returned the max rows that I was expecting.

Reply:
If it's in the constructor it'll only update on creating a new session, correct?

Reply:
it's whenever we create a new future, which currently is on any new query. Should we not be creating a new future every time?